server/src/main/java/org/apache/druid/server/http/SegmentListerResource.java

```diff
@@ -205,7 +209,8 @@ public void onFailure(Throwable th)
               log.debug(ex, "Request timed out or closed already.");
             }
           }
-        }
+        },
```

Comment:
Same. This code looks similar to `TaskManagementResource`. Maybe the common logic should be extracted. Or, there should be comments to both of them noting that these classes are written in a similar way so changes, fixes, refactorings should be applied to both classes at the same time.

Reply:
extract and refactor TaskManagementResource, SegmentListerResource needs more work, will file separate PR to do it
in this PR, will add comments to both classes at the same time

Reply:
Ok, but the comments that you left right here don't make anything clearer. It's ok to omit them in this PR and instead write better comments as part of that PR that you planned, though.

Reply:
sure